src/Categories/Object/NaturalNumbers/Properties/Parametrized.agda

```diff
@@ -0,0 +1,85 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category.Core
+open import Categories.Object.Terminal using (Terminal)
+open import Categories.Category.CartesianClosed.Bundle using (CartesianClosedCategory)
+open import Categories.Category.Cartesian using (Cartesian)
+open import Categories.Category.BinaryProducts using (BinaryProducts)
+open import Categories.Category.Cocartesian using (BinaryCoproducts)
+open import Categories.Category.CartesianClosed using (CartesianClosed)
+
+-- In CCCs NNOs and PNNOs coincide
+
+module Categories.Object.NaturalNumbers.Properties.Parametrized {o ℓ e} (CCC : CartesianClosedCategory o ℓ e) (𝒞-Coproducts : BinaryCoproducts (CartesianClosedCategory.U CCC)) where
+
+open import Level
+
+open CartesianClosedCategory CCC renaming (U to 𝒞)
+open CartesianClosed cartesianClosed using (cartesian; λg; eval′; β′; λ-inj; λ-cong; η-exp′; λ-unique′; subst)
+open Cartesian cartesian using (terminal; products)
+open BinaryProducts products renaming (unique′ to bp-unique′)
+
+open import Categories.Object.NaturalNumbers 𝒞 terminal using (NNO)
+open import Categories.Object.NaturalNumbers.Parametrized cartesianCategory using (ParametrizedNNO)
+open import Categories.Morphism 𝒞
+open import Categories.Morphism.Reasoning 𝒞
+
+open HomReasoning
+open Equiv
+
+open Terminal terminal
+
+NNO×CCC⇒PNNO : NNO → ParametrizedNNO
+NNO×CCC⇒PNNO nno = record 
+  { N = N 
+  ; isParametrizedNNO = record
```

Comment:
Could you please extract these large inline proofs to be in a where clause below the record? This makes downstream usage more comfortable and more efficient.

Reply:
Sure, done!